WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateLogic.java

```diff
@@ -173,13 +170,18 @@ public void run() {
                 localTopics.addAll(ReaderTagTable.getFollowedTags());
                 localTopics.addAll(ReaderTagTable.getBookmarkTags());
                 localTopics.addAll(ReaderTagTable.getCustomListTags());
+                localTopics.addAll(Collections.singletonList(ReaderTag.createDiscoverPostCardsTag()));
+
+                // Check last time we've update tags followed analytics for this user, and force bumping if > 1 day
+                boolean shouldBumpTagsAnalytics = false;
+                long tagsUpdatedTimestamp = AppPrefs.getReaderTagsUpdatedTimestamp();
+                long now = new DateProvider().getCurrentDate().getTime();
+                if (now - tagsUpdatedTimestamp > 1000 * 60 * 60 * 12) { // 12 hrs
+                    shouldBumpTagsAnalytics = true;
+                }
 
-                if (
-                        !localTopics.isSameList(serverTopics)
-                        || displayNameUpdateWasNeeded
-                ) {
-                    AppLog.d(AppLog.T.READER, "reader service > followed topics changed "
-                                              + "updatedDisplayNames [" + displayNameUpdateWasNeeded + "]");
+                if (!localTopics.isSameList(serverTopics)) {
```

Comment:
🚫 I stumbled into this very same issue when investigating the root cause of #20010 and tried fixing it in a similar way to what you did in this PR, but 

A. The implementation here is missing updating `isSameList` to consider the `tagDisplayName` which we update on line 147 above, so this line would return `true` even if the local and server tags of same ids have different names, which is undesirable, especially since we changed the display name for some tags in the recent Reader IA changes, so that means that updating versions wouldn't update the tag names until there's another followed tag change, and 

B. As I mentioned on this [commit](https://github.com/wordpress-mobile/WordPress-Android/commit/3017bab35cb34e8d49b45f092d6ddcae0169014c) in the branch where I am trying to fix #20010 doing this breaks the `Subcscriptions` feed completely (as well as some minor breakage in other feeds based on `ReaderPostListFragment`) because **several** parts of the code are expecting to receive `FollowedTagsChanged` every time the `ReaderUpdateLogic` runs for tags (even when the tags didn't change) to load things like the top bar Filters, loading the suggested tags, and even initialize the Subscriptions feed properly (the function that fetches posts when opening the feed only runs as a result of `FollowedTagsChanged` being triggered.

Scenario B above is the reason I still wasn't able to fix #20010 and I tried a few workarounds to initialize the Subscriptions feed [here](https://github.com/wordpress-mobile/WordPress-Android/compare/issue/20010-reader-ia-subscription-infinite-loading) without success. 

It is really a bummer that fixing this unnecessary writing to the local DB bug makes a lot of things break (because several pieces of logic were made around that bug's side effects), but we will need to figure out a more robust solution for this. 😞 

Here's a video of me testing this PR with an account with 0 followed tags and blogs. Notice how even after subscribing to some tags and blogs, things are still broken.

https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/7d0afe58-5d58-4220-acb3-2f58ea53f403

Some of the broken things from the video:
- **The top bar filters never show up**, since the logic to show them (`SubFilterViewModel#loadSubFilters`) depends on `FollowedTagsChanged` being triggered for some reason...
- **When the user has no blog subscriptions, sometimes the `Subscriptions` feed is just blank**, since the fetching/initialization is done when `FollowedTagsChanged` is triggered and there is logic to hide the "empty view" in `updateCurrentTagIfTime`.
- For the same reason as above, **when unsubscribing/subscribing to a blog on a different screen, the Subscriptions feed is not updated upon entering**, so it displays posts from blogs that are no longer subscribed or doesn't show stuff from blogs I just subscribed, only after a pull to refresh.